tests/testthat/test-expr-expr.R

```diff
@@ -29,3 +29,2946 @@ test_that("map_batches works", {
+  })
+  r_clip <- \(x, a, b) r_clip_min(x, a) |> r_clip_max(b)
+
+  l <- list(
+    int   = c(NA_integer_, -.Machine$integer.max, -4:4, .Machine$integer.max),
+    float = c(NA, -Inf, .Machine$double.xmin, -3:2, .Machine$double.xmax, Inf, NaN)
+  )
+
+  # clip min
+  expect_equal(
+    pl$DataFrame(!!!l)$select(
+      int_mini32 = pl$col("int")$clip(lower_bound = -.Machine$integer.max),
+      int_m2i32 = pl$col("int")$clip(lower_bound = -2L),
+      int_p2i32 = pl$col("int")$clip(lower_bound = 2L),
+      float_minf64 = pl$col("float")$clip(lower_bound = -Inf),
+      # float_NaN2f64 = pl$col("float")$clip(lower_bound = NaN),
+      float_p2f64 = pl$col("float")$clip(lower_bound = 2)
+    ),
+    pl$DataFrame(
+      int_mini32 = r_clip_min(l$int, -.Machine$integer.max),
+      int_m2i32 = r_clip_min(l$int, -2L),
+      int_p2i32 = r_clip_min(l$int, 2L),
+      float_minf64 = r_clip_min(l$float, -Inf),
+      # float_NaN2f64 = l$float,
+      float_p2f64 = r_clip_min(l$float, 2)
+    )
+  )
+
+  # clip max
+  expect_equal(
+    pl$DataFrame(!!!l)$select(
+      int_mini32 = pl$col("int")$clip(upper_bound = -.Machine$integer.max),
+      int_m2i32 = pl$col("int")$clip(upper_bound = -2L),
+      int_p2i32 = pl$col("int")$clip(upper_bound = 2L),
+      float_minf64 = pl$col("float")$clip(upper_bound = -Inf),
+      # float_NaN2f64 = pl$col("float")$clip(upper_bound = NaN),
+      float_p2f64 = pl$col("float")$clip(upper_bound = 2)
+    ),
+    pl$DataFrame(
+      int_mini32 = r_clip_max(l$int, -.Machine$integer.max),
+      int_m2i32 = r_clip_max(l$int, -2L),
+      int_p2i32 = r_clip_max(l$int, 2L),
+      float_minf64 = r_clip_max(l$float, -Inf),
+      # float_NaN2f64 = r_clip_max(l$float, NaN),
+      float_p2f64 = r_clip_max(l$float, 2)
+    )
+  )
+
+  expect_equal(
+    pl$DataFrame(!!!l)$select(
+      a = pl$col("int")$clip(-.Machine$integer.max, -.Machine$integer.max + 1L),
+      b = pl$col("int")$clip(-2L, -2L + 1L),
+      c = pl$col("int")$clip(2L, 2L + 1L),
+      d = pl$col("float")$clip(-Inf, 1),
+      e = pl$col("float")$clip(2, 2 + 1)
+    ),
+    pl$DataFrame(
+      a = r_clip(l$int, -.Machine$integer.max, -.Machine$integer.max + 1L),
+      b = r_clip(l$int, -2L, -2L + 1L),
+      c = r_clip(l$int, 2L, 2L + 1L),
+      d = r_clip(l$float, -Inf, 1),
+      e = r_clip(l$float, 2, 2 + 1)
+    )
+  )
+
+  # clip() accepts strings as column names
+  df <- pl$DataFrame(
+    foo = c(-50L, 5L, NA_integer_, 50L),
+    bound = c(1, 10, 1, 1)
+  )
+  expect_equal(
+    df$select(clipped = pl$col("foo")$clip(lower_bound = "bound")),
+    pl$DataFrame(clipped = c(1L, 10L, NA_integer_, 50L))
+  )
+
+  # clip() works with temporal
+  df <- pl$DataFrame(foo = as.Date(c("2020-01-01", "2020-01-02")))
+  expect_equal(
+    df$select(clipped = pl$col("foo")$clip(lower_bound = pl$lit("2020-01-02"))),
+    pl$DataFrame(clipped = as.Date(c("2020-01-02", "2020-01-02")))
+  )
+})
+
+# TODO-REWRITE: lower bound i32 becomes NA once converted to R
```

Comment:
Don't know how to fix this

Reply:
How about cast to i64?

Reply:
Do you mean in the test or in the implementation?

If we do this in the tests only, the issue is that users might silently run into this int32 -> NA conversion. 
If we do this in the implementation this would add confusion since the input would be int32 but the output would be int64.

Should we have a warning when converting int32 lower bound to R since this is the only case this int32 -> NA conversion would happen?

Reply:
I understand the need for this, but do not plan to address it immediately. So I think we need to cast to String or I64 during testing.
https://github.com/eitsupi/neo-r-polars/blob/1d7ffa8b44966baffc14524d1498938a0cf119f9/src/rust/src/series/export.rs#L68